### PR TITLE
refactor: remove abstraction from the context class

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This pull request removes abstractions from the context class by using properties.

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [ ] I've ensured the change(s) work on `3.8.6` and higher.
- [x] I have added the `versionadded`, `versionchanged` and `deprecated` to any new or changed user-facing function I committed.

### Pull-Request specification

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [ ] To add a new feature
  - [ ] As a general enhancement
  - [x] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/client/context.py [hunk 1]
```diff
@@ -67,16 +66,48 @@ def __attrs_post_init__(self) -> None:
...
+        .. versionchanged:: 4.4.0
+            Channel now returns ``None`` instead of ``MISSING`` if it is not found to avoid confusion
+
+        Returns the current channel.
+        """
+        return self._client.cache[Channel].get(self.channel_id, None)
+
+    @property
+    def guild(self) -> Optional[Guild]:
+        """
```
Comment (mAxYoLo01) [inline interactions/client/context.py:119]: Docstrings should maybe show the difference between `ctx.guild` and `ctx.get_guild()`. Same goes for author and channel?

Comment (sillyElvy) [reply interactions/client/context.py:119]: man that is mean, I wanted you to make it so I could commit it xD

give me a moment

Comment (mAxYoLo01) [reply interactions/client/context.py:119]: hahaha I knew you wanted that, sorry I have assignments to give back xD

File: interactions/client/context.py [hunk 2]
```diff
@@ -75,48 +84,77 @@ def deferred_ephemeral(self) -> bool:
...
-            self.channel = self._client.cache[Channel].get(self.channel_id, MISSING)
+        Returns the current channel, if cached.
+        """
+        return self._client.cache[Channel].get(self.channel_id, None) or self._client.cache[
+            Thread
+        ].get(self.channel_id, None)
+
+    @property
+    def guild(self) -> Optional[Guild]:
+        """
```
Comment (sillyElvy) [inline interactions/client/context.py:119]:
```suggestion
        """
        .. versionadded:: 4.1.0

```

File: interactions/client/context.py [hunk 3]
```diff
@@ -75,48 +84,77 @@ def deferred_ephemeral(self) -> bool:
...
+        Returns the author/member that invoked the interaction.
+        """
+        return self.member
 
-        if self.guild is None and self.guild_id is not None:
-            self.guild = self._client.cache[Guild].get(self.guild_id, MISSING)
+    @property
+    def channel(self) -> Optional[Channel]:
+        """
```
Comment (sillyElvy) [inline interactions/client/context.py:105]:
```suggestion
        """
        .. versionadded:: 4.1.0

```

File: interactions/client/context.py [hunk 4]
```diff
@@ -75,48 +84,79 @@ def deferred_ephemeral(self) -> bool:
...
+        return self._client.cache[Channel].get(self.channel_id, None) or self._client.cache[
+            Thread
+        ].get(self.channel_id, None)
+
+    @property
+    def guild(self) -> Optional[Guild]:
+        """
+        .. versionadded:: 4.1.0
+         .. versionchanged:: 4.4.0
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
         .. versionadded:: 4.1.0

         .. versionchanged:: 4.4.0
```

File: interactions/client/context.py [hunk 5]
```diff
@@ -75,48 +84,79 @@ def deferred_ephemeral(self) -> bool:
...
+        """
+        Returns the author/member that invoked the interaction.
+        """
+        return self.member
+
+    @property
+    def channel(self) -> Optional[Channel]:
+        """
+        .. versionadded:: 4.1.0
+        .. versionchanged:: 4.4.0
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
        .. versionadded:: 4.1.0

        .. versionchanged:: 4.4.0
```

File: interactions/client/context.py [hunk 6]
```diff
@@ -41,11 +43,8 @@ class _Context(ClientSerializerMixin):
     """
 
     message: Optional[Message] = field(converter=Message, default=None, add_client=True)
-    author: Member = field(converter=Member, default=None, add_client=True)
-    member: Member = field(converter=Member, add_client=True)
+    member: Member = field(default=None, converter=Member, add_client=True)  # DMs?
```
Comment (Damego) [inline interactions/client/context.py]: if member is none in the dms should we make it like `Optional[Member]`?

File: interactions/client/context.py [hunk 7]
```diff
@@ -75,48 +84,81 @@ def deferred_ephemeral(self) -> bool:
...
+        return self.id.timestamp
 
-        if self.user is None:
-            self.user = self.member.user if self.member else None
+    @property
+    def author(self) -> Member:
+        """
+        Returns the author/member that invoked the interaction.
+        """
+        return self.member
```
Comment (Damego) [inline interactions/client/context.py:101]: What do you think about make it return `self.user` if member is none?

Comment (sillyElvy) [reply interactions/client/context.py:101]: That's breaking so no

File: interactions/client/context.py (hunk 6 above)
Comment (sillyElvy) [reply interactions/client/context.py]: Sure if you could add a suggestion for it?

File: interactions/client/context.py (hunk 7 above)
Comment (Damego) [reply interactions/client/context.py:101]: okay then

File: interactions/client/context.py (hunk 6 above)
Comment (Damego) [reply interactions/client/context.py]: sure

Comment (Damego) [inline interactions/client/context.py]:
```suggestion
    member: Optional[Member] = field(default=None, converter=Member, add_client=True)
```

File: interactions/client/context.py [hunk 8]
```diff
@@ -75,48 +84,81 @@ def deferred_ephemeral(self) -> bool:
...
 
-        self.author = self.member
+        Returns when the interaction was created.
+        """
+        return self.id.timestamp
 
-        if self.user is None:
-            self.user = self.member.user if self.member else None
+    @property
+    def author(self) -> Member:
```
Comment (Damego) [inline interactions/client/context.py]:
```suggestion
    def author(self) -> Optional[Member]:
```

**Review by Damego — APPROVED** [review]

**Pull request merged**